@xen-orchestra/web/src/utils/fetch.util.ts

```diff
@@ -0,0 +1,59 @@
+import { ApiError } from '@/error/api.error.ts'
+import { HttpCodes } from '@core/types/http-codes.type.ts'
+
+const DEFAULT_OPTIONS: RequestInit = {
+  headers: {
+    'Content-Type': 'application/json',
+  },
+  credentials: 'same-origin',
+}
+
+export async function fetchRequest<T>(endpoint: string, options: RequestInit = {}): Promise<T> {
+  const requestOptions: RequestInit = {
+    ...DEFAULT_OPTIONS,
+    ...options,
+    headers: {
+      ...DEFAULT_OPTIONS.headers,
+      ...(options?.headers || {}),
+    },
+  }
+
+  try {
+    const response = await fetch(endpoint, requestOptions)
+
+    let data
```

Comment:
data is of type `any`
```suggestion
    let data: T | Record<string, unknown> | undefined 
```